src/subcommand/wallet/burn.rs

```diff
@@ -141,4 +161,143 @@ impl Burn {
       .build_transaction()?,
     )
   }
+
+  fn create_unsigned_burn_runes_transaction(
```

Comment:
I definitely think we should deduplicate this with send, it's a long function, and they seem really close to each other.